synapse/federation/federation_server.py

```diff
@@ -469,14 +476,26 @@ async def _handle_pdus_in_txn(
                 logger.info("Ignoring PDU: %s", e)
                 continue
 
-            event = event_from_pdu_json(p, room_version)
+            try:
+                event = event_from_pdu_json(p, room_version)
+            except Exception as e:
+                # We can only provide feedback to the federating server if we can determine what the event_id is
+                # but since we failed to parse the event, we can't derive the `event_id` so there is nothing
+                # to use as the `pdu_results` key. Best we can do is just log for our own record and move on.
```

Comment:
Just re-wrapping to fit better:

```suggestion
                # We can only provide feedback to the federating server if we can
                # determine what the `event_id` is but since we failed to parse the
                # event, we can't derive the `event_id` so there is nothing to use as
                # the `pdu_results` key. Best we can do is just log for our own record
                # and move on.
```